### ISSUE TITLE
Which flash do I need to select in the software

### ISSUE THREAD

**Issue by tomse** [OlegMishin/FlashROM#5]
TL866-II plus
I've tried TSOP48 and SOP44
both gives pin errors.

TL866 (using downgraded firmware as mentioned on proghq)
doesn't seem to enable "WE" as writes output but it's never saved in the flash.
I can fill with FF's and it gives no errors - when it's already empty.
trying to program something gives instant verify error, and nothing is saved on the flash.

I've made 2 adapters and 2 flash roms 40 pin, using ST29f800
installed the flash and adapter as shown on the bottom picture.

**Comment by OlegMishin** [comment]
Hi,
You need to select in TL866 software exact Flash part number as your Flash IC is. I don't see ST29f800 in device support list. Is it M29F800AB or similar?
I hope you flashed ATtiny micro in the adapter, otherwise it will not work.

I use TL866II plus with the latest stock firmware. Maybe the downgrade is an issue.

**Comment by tomse** [comment]
Yes it is  the ST M29F800FB (sorry my bad for not adding the M), pincompatible with M29F800AB.

I was somewhat confused as to which version and programmer to use. You link to a different site where the older software/firmware is said to be used and using the First release of the TL866. Luckily I have both versions.. and using the TL866II Plus with latest software and firmware (right now it's 12.63).

Hooked up like the documentation. ATTiny is flashed. using the 40pin flash.

flashing using the ST29F800AB TSOP48 device (FB doesn't show on the list, but it should be fine as they're pin compatible).

Getting this result.
![image](https://github.com/OlegMishin/FlashROM/assets/1444539/f251707f-b7dc-40ed-8f78-f384fc79182d)


There are no pin1 markers on my ATTiny, but I have the text turned the same way as the 2x74HCT373's
J1 does that need to be closed?
and when should the diode light up? (so far it hasn't)

**Comment by OlegMishin** [comment]
Strange, M29F800FB  is in supported devices in the SW I use:
![image](https://github.com/OlegMishin/FlashROM/assets/81614352/722c9f8f-666b-40b0-b818-140610063e1b)
"Bad pins" error means there is a problem with soldering or connections. You need to check soldering of the flash chip and connections of pins A17(pin32 of ZIF socket) and BYTE#(pin40) marked in red on the picture.

**Comment by tomse** [comment]
I have a version of the original adapter shown above.. this one works, so I've eliminated the programmer as being faulty.
I have 2 adapter boards and 2 flashroms  intermixing these doesn't deviate from the pins that gives errors. I should be very unlucky if I made bad solderings on these boards to give the same error :-P  (I did go through them with a microscope to eliminate bad soldering).

I think you missed my 3 questions beneath the image

**Comment by OlegMishin** [comment]
Sorry, missed in deed. ATtiny's orientation is correct. JP1 needs to be closed. The led lits when power supply provided to the flash, when any operation is ongoing(read/write etc).

**Comment by tomse** [comment]
I found the issue pin issues... I didn't close the jumpers on the flash chips - now I've shorted J1 and J2 for Rev6
(perhaps this needs to be added to the documentation / Q&A?)

now I'm down to data isn't being written to the chip.
tried both adapters and both flash chips - something else I might have missed?
![image](https://github.com/OlegMishin/FlashROM/assets/1444539/e0ed5b43-751a-4965-821c-02224ada6d2b)

**Comment by OlegMishin** [comment]
If you enable "Check ID" does it work? Which ID actually read from the flash?
Even if M29F800FB is pin compatible with M29F800AB it doesn't mean that programming algorithm is the same.
You can try to select M29F800FB from Numonix.

**Comment by tomse** [comment]
the ID says 0xFFFF FFFF so unidentified. I bought the IC's at tme.eu, so I'm not expecting them to be some chinese knock-offs.

I have some IC's that I've tried programming, erasing etc using the original TSOP adapter they work fine - ic's are from the same batch bought from tme.  

Erasing the IC 40pin rom adapter takes 0.016ms, whereas erasing using the TSOP adapter takes 15 seconds, so something is off..
can't get my head around as of why.. 

can it be the attiny that causes this?
I just did a reprogram on the tiny, checking fuses, hex file etc, no errors while programming it

**Comment by OlegMishin** [comment]
To ckeck if the ATtiny is the bad guy, you can remove R1 and R2 and check. I guess the SW should report an adapter error. 
![image](https://github.com/OlegMishin/FlashROM/assets/81614352/7a2149c7-da38-4a62-90ca-1576befc71be)

Just to confirm, A18, BSY, RST and WE wires between adapter and FlashROM are on their place, right?

**Comment by tomse** [comment]
yes... the 4 cables are connected to their respective connectors.

getting bad pins when the resistors are removed.
![image](https://github.com/OlegMishin/FlashROM/assets/1444539/64485da6-0d18-4e4e-a4ef-35e02d7cac58)


and just an adapter error when "Pin detect" are removed (trying to run the Erase function

**Comment by OlegMishin** [comment]
Sorry, no more ideas.... I have never had any issue with my adapter using many different Flash ICs.
Looks like a soldering problem or pcb defekt. Have you built only one?

**Comment by tomse** [comment]
I built 2 adapters, and 2 flash roms (40pins)

thanks for the help though :-)

I'll leave the issue open, that I can make a closing comment if I find the error.. otherwise I'll close it on a later time as unreproducable..
is that ok with you?
